### PR TITLE
Tutorial: move from title screen to campaign menu

### PR DESCRIPTION
**Pull request by nemaara**
As discussed in #4543. Tagging @stevecotton for comments.

### CODE REVIEW

**Comment by nemaara** [comment]
Note that this requires the corresponding C++ to be deleted, but I don't know exactly everything that needs to go away (probably something in title_screen.cpp/.hpp) so I'll add that to the TODO list on #4543 when this is done.

**Comment by CelticMinstrel** [comment]
Probably some stuff in game_config_manage.cpp and wesnoth.cpp needs to be deleted too, basically anything that treats the tutorial as a special case.

Then we'll also want to add something like `rank=tutorial` as mentioned elsewhere.

**Comment by stevecotton** [comment]
Looks good to me.

> Note that this requires the corresponding C++ to be deleted

The C++ will handle this (merged in PR #4552). Some clean-up could be done to drop the unused code, but it's not a blocker.

File: data/campaigns/tutorial/_main.cfg [hunk 1]
```diff
@@ -6,6 +6,31 @@
...
 
+[campaign]
+    id=Tutorial
+    name= _ "Battle Training"
+    icon="data/campaigns/tutorial/images/units/quintain.png~RC(magenta>blue)"
+    image="data/campaigns/tutorial/images/units/quintain.png~RC(magenta>blue)~XBRZ(4)"
+    abbrev= _ "Tutorial"
+    rank=1
+    # put it last when sorted by date
+    year="9999 AF"
```
Comment (sigurdfdragon) [inline data/campaigns/tutorial/_main.cfg]: year=519 YW
is also an option (right before HttT)

**Comment by sigurdfdragon** [comment]
Adding the scenario count in the description might get some more people to look at it and think "oh there's something here" instead of "blah, tutorial."
Also saying it is 2 scenarios might help entice new players when they compare it to AToTB's 4 scenarios & higher for other campaigns (assuming the tutorial stays at 2 scenarios)

File: data/campaigns/tutorial/_main.cfg (hunk 1 above)
Comment (nemaara) [reply data/campaigns/tutorial/_main.cfg]: The tutorial isn't canonical lore-wise, so I think either first or last would be better.

**Comment by nemaara** [comment]
I purposely opted out of giving the scenario count because the tutorial isn't really 2 full scenarios, and it doesn't make sense to count it the same way as for campaigns (incidentally, this is why I initially didn't really want to put it in the campaign menu, but probably this is the best fix for now). Also, that might change pretty soon if we do end up deciding to merge AOI into the tutorial.

**Comment by sigurdfdragon** [comment]
> the tutorial isn't really 2 full scenarios, and it doesn't make sense to count it the same way as for campaigns

I think new players would see it as: "I played on a map and had an objective, then went to a different map with a different objective, which was tied well enough to the first map, and I got to recall troops on the 2nd map. Therefore it's 2 scenarios."

**Comment by shikadiqueen** [comment]
I have never agreed with this "we must tell the players everything upfront" mentality that has been plaguing Wesnoth for the past 10 years (which also resulted in the addition of the "this is the last scenario" objective note).

Part of the experience of entertainment is not knowing the destination most of the time. If you are really invested into a book you aren't going to constantly check your progress. If you are really invested into a movie in the cinema you aren't going to grab your phone and check what time it is and see how long it's been since the movie started. If you are really invested into a game you aren't going to keep count of the level number you are on and how many levels long it is. This is why I ended up deliberately lying to my players on campaign length in my own campaigns by stretching the definition of what constitutes a "scenario" as needed to suit my own ends.

But I digress.

Ultimately, players are going to decide to play the tutorial depending on two factors:

1. Whether the game has sold itself well enough for them to have gone through the hassle of finding the Store page or download link that led them to installing it.
2. Whether they are new to the game or not (this includes the possibility of having watched a YTer/streamer play it) and feel they need to get the hang of it in a Tutorial environment instead of playing a proper campaign.

The tutorial is meant to be pretty short by nature. It's never going to be long enough to bother any player — and if this somehow stops being the case that means we've failed as a **game** development team. In most players' mind, the tutorial's length is never going to be a decisive factor in whether they are going to play it or not and we should not give them a reason for it to be so.

**Comment by stevecotton** [comment]
Let's give the player information useful for deciding where to start.

"Unless you've either played Wesnoth before or have watched others play, skipping the tutorial is not recommended. Some of the basic parts of the game aren't expected to be discovered through button-mashing - for example the context menu of castle hexes has extra options when your leader is standing on a connected keep. The basics that are introduced in the tutorial are expected to be known by anyone playing the rookie campaigns."

**Comment by sigurdfdragon** [comment]
My point of listing the scenario count for the tutorial was that it would be an extra encouragement to play it when new players compare it to the campaigns. If it's not thought to be such, or a better way of encouraging players is found, that's fine.

**Comment by nemaara** [comment]
Merging later today if there aren't any more comments.

**Pull request merged**